src/clj/game/core/ice.clj

```diff
@@ -219,17 +219,13 @@
   (update! state :corp (dont-resolve-all-subroutines ice)))
 
 (defn reset-sub
-  [ice sub]
-  (assoc ice :subroutines (assoc (:subroutines ice) (:index sub) (dissoc sub :broken :fired :resolve))))
-
-(defn reset-sub!
-  [state ice sub]
-  (update! state :corp (reset-sub ice sub)))
+  [sub]
+  (dissoc sub :broken :fired :resolve))
```

Comment:
I would prefer to keep the api the same as before. Every sub manipulation function takes ice and sub, so this one being different isn't great.

Given that `:index` isn't being touched by this function, why shouldn't we updating it on the ice directly? When will resetting a sub change its indexes?

(Same question applies to reset-all-subs too.)

Reply:
The problem arose here when you ended up with something like: 

```
{:subroutines ({:index 1 ...})}
```

With the associating directly into the subroutines list via list index, this actually did a copy rather than an update, hence why the `Trash this ICE` sub was being duplicated.

To me the top level function looks like it just wants to clear the resolved/broken/fired flags on all subroutines for the ice, this seems completely like something that should just apply to everything in the list and feels more like a straight up `map` of something rather than something that has to be concerned about the stored indexes being correct.

Reply:
I would actually be fine with just inlining and deleting `reset-sub` as well since it's currently only used in `reset-all-subs`.

Reply:
Deleting the function is fine with me.

Reply:
Cool - I still declared it separately via `leftfn` which names it a bit (it was hard for me in the old code to figure out what was happening) but also because my clojure-fu is weak and I wasn't sure how to do the map function inside the inline function

Reply:
Perfect use of letfn.